adl/rot13adl/rot13prototypes.go

```diff
@@ -0,0 +1,31 @@
+package rot13adl
+
+// Prototype embeds a NodePrototype for every kind of Node implementation in this package.
+// This includes both the synthesized node as well as the substrate root node
+// (other substrate interior node prototypes are not exported here;
+// it's unlikely they'll be useful outside of the scope of the ADL's implementation package.)
+//
+// You can use it like this:
+//
+// 		rot13adl.Prototype.Node.NewBuilder() //...
+//
+// and:
+//
+// 		rot13adl.Prototype.SubstrateRoot.NewBuilder() // ...
+//
+var Prototype prototype
+
+// This may seem a tad mundane, but we do it for consistency with the way
+// other Node implementation packages (like basicnode) do this:
+// it's a convention for "pkgname.Prototype.SpecificThing.NewBuilder()" to be defined.
+
+type prototype struct {
+	Node          _R13String__Prototype
+	SubstrateRoot _Substrate__Prototype
+}
+
+// REVIEW: In ADLs that use codegenerated substrate types defined by an IPLD Schema, the `Prototype.SubstrateRoot` field...
```

Comment:
Here's another of the big review questions: what should we expect about substrate nodes and their prototypes and builders? (... In ADLs that export some! -- remember, this is not strictly mandated; ADLs can be implemented using just basicnode in their guts too.)

There's some disharmony with how schema typed nodes currently treat this.  I'm not sure in which direction this should be resolved -- by moving the ADLs closer to how typed nodes behave?  Or by changing how schema nodes behave to be more like ADLs, returning the same form you started with and making you take another step to get high level views out?